src/rcc.rs

```diff
@@ -204,6 +227,17 @@ impl APB4 {
     }
 }
 
+/// RCC Domain 3 Kernel Clock Configuration Register
+pub struct D3CCIPR {
+    _0: (),
+}
+
+impl D3CCIPR {
+    pub(crate) fn constrain(&mut self) -> &rcc::D3CCIPR {
+        unsafe {&(*RCC::ptr()).d3ccipr}
```

Comment:
The logic looks good, but I think the name "constrain" is misleading here. This word is used to mean a function that consumes the structure it is called on, and returns some other type that is a constrained representation. Eg. after calling `constrain` on `PWR` the user's code can't access `PWR` again. This function is nothing like that - it's an accessor to say "it's okay to modify D3CCIPR within this crate at any time".

Something like "kernel_clock_configuration_input" or "kernel_ccip" would be better